src/game/audio/soundtrack.rs

```diff
@@ -2,43 +2,49 @@ use bevy::{audio::PlaybackMode, prelude::*};
 
 use crate::game::assets::{HandleMap, SoundtrackKey};
 
-pub(super) fn play_soundtrack(
-    trigger: Trigger<Soundtrack>,
+pub(super) fn plugin(app: &mut App) {
+    app.register_type::<IsSoundtrack>();
+    app.observe(play_soundtrack);
+}
+
+fn play_soundtrack(
+    trigger: Trigger<PlaySoundtrack>,
     mut commands: Commands,
     soundtrack_handles: Res<HandleMap<SoundtrackKey>>,
-    query: Query<Entity, With<SoundtrackMarker>>,
+    soundtrack_query: Query<Entity, With<IsSoundtrack>>,
 ) {
-    let event = trigger.event();
-    for entity in query.iter() {
-        commands.entity(entity).despawn();
-    }
-
-    let source = match event {
-        Soundtrack::Disable => {
-            return;
-        }
-        Soundtrack::Credits => &soundtrack_handles[&SoundtrackKey::Credits],
-        Soundtrack::Gameplay => &soundtrack_handles[&SoundtrackKey::Gameplay],
+    for entity in &soundtrack_query {
+        commands.entity(entity).despawn_recursive();
     }
-    .clone_weak();
 
-    let settings = PlaybackSettings {
-        mode: PlaybackMode::Loop,
-        ..default()
-    };
-    commands.spawn((AudioSourceBundle { source, settings }, SoundtrackMarker));
+    commands.spawn((
+        AudioSourceBundle {
+            source: soundtrack_handles[&match trigger.event() {
```

Comment:
Same as above in similar code

Reply:
Split getting the key out into an earlier step.